tests/functional/codegen/features/iteration/test_for_range.py

```diff
@@ -145,26 +228,28 @@ def foo(a: {typ}) -> {typ}:
     assert c.foo(100) == 31337
 
 
-# test that we can get to the upper range of an integer
 @pytest.mark.parametrize("typ", ["uint8", "int128", "uint256"])
 def test_for_range_edge(get_contract, typ):
+    """
+    Check that we can get to the upper range of an integer.
+    Note that to avoid overflow, given the semantics of range(),
```

Comment:
hmm is this a problem with the new implementation of `range(x, y, bound=)`? like we can't get to `max_value()` for a given type?

Reply:
It was not possible to use the format `range(x, y, bound=N)` but only `range(x, x + 1)`. That was because we totally ignored the `x+1` and only looked at the `+1` part:
https://github.com/vyperlang/vyper/blob/c6f457a73db40e4b113497883bd330e0dcec28d1/vyper/codegen/stmt.py#L270-L271

max_value + 1 would otherwise always overflow

Reply:
ah, i see.

Reply:
```suggestion
    Note that to avoid overflow in the bounds check for range(),
```